scripts/linters/js_ts_linter.py

```diff
@@ -859,6 +857,8 @@ def perform_all_lint_checks(self):
                 'There are no JavaScript or Typescript files to lint.')
             return []
 
+        clean.delete_directory_tree('tmpcompiledjs/')
```

Comment:
Should probably add single-line comment explaining the "why" for this, as well as the corresponding call below. Also add a single line comment above _compile_ts() explaining that it will create and populate this dir.

Reply:
Done